proto/substrait/algebra.proto

```diff
@@ -760,6 +765,53 @@ message NestedLoopJoinRel {
     JOIN_TYPE_RIGHT_ANTI = 8;
     JOIN_TYPE_LEFT_SINGLE = 9;
     JOIN_TYPE_RIGHT_SINGLE = 10;
+    JOIN_TYPE_MARK = 11;
+  }
+
+  substrait.extensions.AdvancedExtension advanced_extension = 10;
+}
+
+message DuplicateEliminatedGetRel{
+   RelCommon common = 1;
+   ReferenceRel input = 2;
+   repeated Expression.FieldReference column_ids = 3;
+}
+
+message DuplicateEliminatedJoinRel {
+  RelCommon common = 1;
+  Rel left = 2;
+  Rel right = 3;
+
+  Expression expression = 4;
+  Expression post_join_filter = 5;
+
+  JoinType type = 6;
+
+  // The set of columns that will be duplicate eliminated from the LHS and pushed into the RHS
+  repeated Expression.FieldReference duplicate_eliminated_columns = 7;
+
+  DuplicateEliminatedSide duplicate_eliminated_side = 8;
+
+  // If this is a DuplicateEliminatedJoin, whether it has been flipped to de-duplicating the LHS or RHS
+  enum DuplicateEliminatedSide {
+      DUPLICATE_ELIMINATED_SIDE_UNSPECIFIED = 0;
+      DUPLICATE_ELIMINATED_SIDE_LEFT = 1;
+      DUPLICATE_ELIMINATED_SIDE_RIGHT = 2;
+    }
+
+ enum JoinType {
```

Comment:
JoinType is probably not needed with DuplicateEliminatedSide here.

Reply:
Hi David, I don't think I understand the comment. The Duplicate Eliminated Join is basically a Logical Comparison Join and has a Join Type.
Or do you mean the enum definition should come right below `JoinType type = 6;`? 
Or that it should reference it differently?

Reply:
If this has all of those capabilities then we are missing the join comparison and maybe some other fields from the standard JoinRel.  If there is a possibility for a left anti join with the right side deduplicated from than the others then the join type is definitely needed.

I was considering this physical because it was a specific way of handling a join.  But it sounds like this is instead another dimension over the existing JoinRel that provides duplicate elimination.  I don't know if that means it should just be more options there in that case or if the separate approach is fine but we can discuss that on the main repo with the others.

Reply:
Indeed, it might be the case that we just extend JoinRel.

For clarity/reference, in DuckDB the Duplicate Eliminated Join, is literally the [LogicalComparisonJoin](https://github.com/duckdb/duckdb/blob/main/src/include/duckdb/planner/operator/logical_comparison_join.hpp#L27) with a `LogicalOperatorType::LOGICAL_DELIM_JOIN`.

The possible join types for the l[ogical operator type](https://github.com/duckdb/duckdb/blob/main/src/include/duckdb/common/enums/logical_operator_type.hpp#L45) are:
```
LOGICAL_JOIN = 50,
LOGICAL_DELIM_JOIN = 51,
LOGICAL_COMPARISON_JOIN = 52,
LOGICAL_ANY_JOIN = 53,
LOGICAL_CROSS_PRODUCT = 54,
LOGICAL_POSITIONAL_JOIN = 55,
LOGICAL_ASOF_JOIN = 56,
LOGICAL_DEPENDENT_JOIN = 57
```

The comparison Join only uses the following types:
```
LOGICAL_DELIM_JOIN = 51,
LOGICAL_COMPARISON_JOIN = 52,
LOGICAL_ASOF_JOIN = 56,
LOGICAL_DEPENDENT_JOIN = 57
```
I think we could potentially add an enum to joinrel with only `LOGICAL_DELIM_JOIN` and `LOGICAL_COMPARISON_JOIN` for now.